### PR TITLE
feat(catalogs): add a few catalog listing endpoints

### PR DESCRIPTION
**Pull request by wweaver**
## What is the motivation for this pull request?
Add initial catalogs functionality

https://www.braze.com/docs/api/endpoints/catalogs

## What is the current behavior?
no catalogs functionality

## What is the new behavior?
Added initial catalog GET functionality

## Checklist:

- [x] [Conventional Commits](https://www.conventionalcommits.org/)
- [x] Tests
- [x] Documentation

### CODE REVIEW

File: src/catalogs/synchronous/catalogs.test.ts [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+  const options = {
+    headers: {
+      Authorization: `Bearer ${apiKey}`,
+      'Content-Type': 'application/json',
+    },
+  }
+
+  describe('List Catalogs', () => {
+    const response: CatalogListResponse = { catalogs: [], message: 'success' }
+    it('with no params', async () => {
```
Comment (remarkablemark) [inline src/catalogs/synchronous/catalogs.test.ts:31]: Nit: could we update the test description so it's more readable here and below?

```suggestion
    it('is called with no params', async () => {
```

File: src/catalogs/synchronous/catalogs.test.ts [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+  const braze = new Braze(apiUrl, apiKey)
+  const options = {
+    headers: {
+      Authorization: `Bearer ${apiKey}`,
+      'Content-Type': 'application/json',
+    },
+  }
+
+  describe('List Catalogs', () => {
+    const response: CatalogListResponse = { catalogs: [], message: 'success' }
```
Comment (remarkablemark) [inline src/catalogs/synchronous/catalogs.test.ts:30]: Nit: could you add a newline so there's separation between the variable and test code block?

File: src/templates/content-blocks/content_blocks.test.ts [hunk 3]
```diff
@@ -19,28 +20,82 @@ describe('Content Blocks', () => {
...
+      description: '',
+      inclusion_count: 0,
+      inclusion_data: [],
+      last_edited: '',
+      name: '',
+      tags: [],
+      message: 'success',
+    }
+
+    it('with required params', async () => {
```
Comment (remarkablemark) [inline src/templates/content-blocks/content_blocks.test.ts:39]: Nit: update test description here and below:

```suggestion
    it('is called with required params', async () => {
```

**Review by remarkablemark — CHANGES_REQUESTED** [review]
Thanks for opening this PR and it looks good overall, left some nitpick comments

File: src/catalogs/synchronous/catalogs.test.ts (hunk 1 above)
Comment (wweaver) [reply src/catalogs/synchronous/catalogs.test.ts]: updated

File: src/templates/content-blocks/content_blocks.test.ts (hunk 3 above)
Comment (wweaver) [reply src/templates/content-blocks/content_blocks.test.ts]: updated

File: src/catalogs/synchronous/catalogs.test.ts (hunk 2 above)
Comment (wweaver) [reply src/catalogs/synchronous/catalogs.test.ts]: updated

**Review by remarkablemark — APPROVED** [review]
LGTM @wweaver

Note for future: if you're updating tests, feel free to use the `test` type in your [commit message](https://www.conventionalcommits.org/en/v1.0.0/) (since `fix` will show up as a bugfix in `CHANGELOG.md`). Thanks!

**Comment by remarkablemark** [comment]
Release:

- https://github.com/remarkablemark/braze-api/pull/187#issuecomment-1485983547

**Comment by wweaver** [comment]
Thanks @remarkablemark. Sorry I don't commit to public repos too often and haven't officially used conventional commits anywhere before.

**Comment by wweaver** [comment]
In the future if I add more code to this I'm not against amending / rebasing to get the commit messages appropriate.

**Comment by remarkablemark** [comment]
All good @wweaver! I appreciate your effort in making this open source package better. Also, would you be interested in being a maintainer?

**Comment by wweaver** [comment]
> All good @wweaver! I appreciate your effort in making this open source package better. Also, would you be interested in being a maintainer?

Sure @remarkablemark. What would that entail? Reviewing others code?

**Comment by wweaver** [comment]
Works for me @remarkablemark. Thanks!

**Comment by wweaver** [comment]
Do you want me to still work on my fork to create branches or create directly on this project when that happens?

**Pull request merged**